solidity/contracts/bridge/Wallets.sol

```diff
@@ -386,13 +400,35 @@ library Wallets {
         self.liveWalletsCount--;
     }
 
-    /// @notice Closes the given wallet and notifies the ECDSA registry
-    ///         about this fact.
+    /// @notice Begins the closing period of the given wallet.
     /// @param walletPubKeyHash 20-byte public key hash of the wallet
     /// @dev Requirements:
     ///      - The caller must make sure that the wallet is in the
-    ///        Live or MovingFunds state.
-    function closeWallet(
+    ///        MovingFunds state
+    function beginWalletClosing(
+        BridgeState.Storage storage self,
+        bytes20 walletPubKeyHash
+    ) internal {
+        Wallet storage wallet = self.registeredWallets[walletPubKeyHash];
+        // Initialize the closing period.
+        wallet.state = WalletState.Closing;
+        /* solhint-disable-next-line not-rely-on-time */
+        wallet.closingStartedAt = uint32(block.timestamp);
+
+        emit WalletClosing(wallet.ecdsaWalletID, walletPubKeyHash);
+    }
+
+    /// @notice Finalizes the closing period of the given wallet and notifies
+    ///         the ECDSA registry about this fact.
+    /// @param walletPubKeyHash 20-byte public key hash of the wallet
+    /// @dev Requirements:
+    ///      - The caller must make sure that the wallet is in the Closing state
```

Comment:
If we are doing a transition between states, then I think `Wallets.sol` needs to check the initial state. But let's see how this ends up in practice at the very end when we have all the building pieces in place.

Reply:
The initial state checked by the caller before calling the transition function is the in-force convention in the `Wallets` lib at the moment. I agree we should revisit it since a lot of things changed since that convention was introduced.